evalml/objectives/utils.py

```diff
@@ -214,3 +229,187 @@ def get_core_objectives(problem_type):
+        max_objectives (dict[str,float]): The mapping of objectives to the maximum values for normalization
+        min_objectives (dict[str,float]): The mapping of objectives to the minimum values for normalization
+
+    Returns:
+        A dictionary mapping objective names to their new normalized values
+    """
+    normalized = {}
+    for objective_name, val in objectives_to_normalize.items():
+        objective_obj = get_objective(objective_name)
+        # Only normalize objectives that are not bounded like percentages
+        # R2 also does not get normalized as it's essentially bounded like a percentage,
+        # and we want to penalize aggressively when R2 is negative
+        if objective_obj.is_bounded_like_percentage or objective_obj.name == "R2":
+            normalized[objective_name] = val
+            continue
+        max_val, min_val = (
+            max_objectives[objective_name],
+            min_objectives[objective_name],
+        )
+        if max_val == min_val:
+            normal = 1
+        else:
+            normal = (val - min_val) / (max_val - min_val)
+            if not objective_obj.greater_is_better:
+                normal = 1 - normal
+        normalized[objective_name] = normal
+    return normalized
+
+
+def recommendation_score(
+    objectives,
+    prioritized_objective=None,
+    prioritized_weight=0.5,
+    custom_weights=None,
+):
+    """Computes a recommendation score for a model given scores for a group of objectives.
+
+    This recommendation score is a weighted average of the given objectives, by default all weighted equally. Passing
+    in a prioritized objective will weight that objective with the prioritized weight, and all other objectives will
+    split the remaining weight equally.
+
+    Args:
+        objectives (dict[str,float]): A dictionary mapping objectives to their values. Objectives should be a float between
+            0 and 1, where higher is better. If the objective does not represent score this way, scores should first be
+            normalized using the normalize_objectives function.
+        prioritized_objective (str): An optional name of a priority objective that should be given heavier weight
+            than the other objectives contributing to the score. Defaults to None, where all objectives are
+            weighted equally.
+        prioritized_weight (float): The weight (maximum of 1) to attribute to the prioritized objective, if it exists.
+            Should be between 0 and 1. Defaults to 0.5.
+        custom_weights (dict[str,float]): A dictionary mapping objective names to corresponding weights between 0 and 1.
+            If all objectives are listed, should add up to 1. If a subset of objectives are listed, should add up to less
+            than 1, and remaining weight will be evenly distributed between the remaining objectives. Should not be used
+            at the same time as prioritized_objective.
```

Comment:
It's my fault for doing this piecemeal - I blame being distracted - but perhaps we should consider an API where these two are put together and that the input is a dict of str:float.  If there is one value, then perhaps that should be interpreted as the prioritized weight/obj.

Reply:
As we've discussed previously, I'm hesitant to completely remove the prioritized objective/weight API. I think the benefit of the simplicity outweighs the slightly larger API. That being said, after thinking about it more I would be open to removing the `prioritized_weight` argument, and just keeping `prioritized_objective`. That allows users to have an easy way to say "I care about this one more", and if they want to be more specific about the weights, there is still `custom_weights` to open up that opportunity. Thoughts?